components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -279,6 +281,50 @@ private static boolean isRoleAlreadyApplied(String username, String roleName, Us
         return isRoleAlreadyApplied;
     }
 
+    public static List<ApplicationBasicInfo> filterAuthorizedApplicationBasicInfo(
+            ApplicationBasicInfo[] applicationBasicInfos, String username)
+            throws IdentityApplicationManagementException {
+
+        List<ApplicationBasicInfo> appInfo = new ArrayList<>();
+        UserStoreManager userStoreManager;
+        String[] userRoles = new String[0];
+        try {
+            userStoreManager = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getUserStoreManager();
+            if (!(userStoreManager instanceof AbstractUserStoreManager)) {
+                userRoles = userStoreManager.getRoleListOfUser(username);
+            }
+        } catch (UserStoreException e) {
+            throw new IdentityApplicationManagementException("Error while loading the user store manager: ", e);
+        }
+
+        for (ApplicationBasicInfo applicationBasicInfo : applicationBasicInfos) {
+            boolean isAllowedApp = false;
+            String applicationRoleName = getAppRoleName(applicationBasicInfo.getApplicationName());
+            if (log.isDebugEnabled()) {
+                log.debug("Checking whether user has role : " + applicationRoleName +
+                        " by retrieving role list of user : " + username);
+            }
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager &&
+                        ((AbstractUserStoreManager) userStoreManager).isUserInRole(username, applicationRoleName)) {
+                    isAllowedApp = true;
+                    appInfo.add(applicationBasicInfo);
+                } else if (Arrays.asList(userRoles).contains(applicationRoleName)) {
+                    isAllowedApp = true;
+                    appInfo.add(applicationBasicInfo);
+                }
```

Comment:
Shall we simplify this logic as follows:
```
if ((userStoreManager instanceof AbstractUserStoreManager &&
                          ((AbstractUserStoreManager) userStoreManager).isUserInRole(username, applicationRoleName))
                      || Arrays.asList(userRoles).contains(applicationRoleName)) {
                  if (log.isDebugEnabled()) {
                      log.debug("Retrieving basic information of authorized application: " +
                              applicationBasicInfo.getApplicationName() + " of the user: " + username);
                  }
                  appInfo.add(applicationBasicInfo);
              }
```

Reply:
Fixed by - https://github.com/wso2/carbon-identity-framework/pull/4394/commits/5ffd769e055b92ac307c24ca4afe1c2688554b09